Dockerfile

```diff
@@ -310,10 +310,29 @@ RUN chown -R $USER /work/rust-optee-trustzone-sdk/optee-qemuv8-3.7.0/build/build
 WORKDIR /
 RUN ./cleanup.sh
 
+FROM build_sgx as build_icecap
+
+ARG USER
+ARG UID
+ENV DEBIAN_FRONTEND noninteractive
+SHELL ["/bin/bash", "-c"]
+
+USER root
+RUN apt-get update && apt-get install --no-install-recommends -y bc
+USER $USER
+
+ENV PATH="/nix/env/bin:${PATH}"
+ENV MANPATH="/nix/env/share/man:${MANPATH}"
+ENV NIX_SSL_CERT_FILE=/nix/env/etc/ssl/certs/ca-bundle.crt
+
+# HACK for persistent git cache. $XDG_CACHE_HOME doesn't appear to be in use elsewhere.
+# TODO patch nix to use $NIX_CACHE_HACK instead
+ENV XDG_CACHE_HOME=/nix/cache
+
```

Comment:
Wouldn't it be better to have these in an Icecap-specific Dockerfile, similar to the rest of the supported platforms?

Reply:
I still haven't rebased this on top of Eric's splitting. Doing so will address this comment and you other comment about SGX as a base.